### DISCUSSION TITLE
QWeather source

### DISCUSSION THREAD

**Discussion by OSNewbie** [breezy-weather/breezy-weather#1170] [General]
Weather API interface for Chinese users, support for free and paid two packages paid, there are corresponding development documentation

[Wabi-sabi weather
](https://www.qweather.com/en/)

**Comment by papjul** [comment]
Unfortunately, QWeather requires your personal phone number, which goes against the privacy goals of the app.

You can find an archived branch `qweather` if you want to build your own version. It will need rebase and it's untested because no maintainer wanted to give their phone number to get a key.

**Reply by OSNewbie** [reply]
You can try email registration or Google login, but then, you can have the contributor provide the key if you don't want to Use the limit for the number of requests, it shouldn't be that high, after all, the version must be partially restricted

**Reply by OSNewbie** [reply]
Regarding the point of collecting user information, this is a flaw in the Chinese application, and there is no way to improve it for the time being.

**Reply by OSNewbie** [reply]
I can provide the weatherkey but I won't access it because I don't understand it so you can test it first, I'm just your average hacker and don't really read dev docs

**Comment by OSNewbie** [comment]
That Color Cloud Weather is not the same as the China Meteorological Administration one, is it also accessed via API? Not all of them are applicable to China Most of the weather APIs that you want to access may have some limitations in terms of registration.

**Comment by OSNewbie** [comment]
I just want the user to be able to customize the weather source, but I feel it's very difficult to implement, after all, you can't adapt to multiple countries at the same time, this developer increases the difficulty

**Comment by papjul** [comment]
I think we have some understanding issues through translation.

It's not an issue with the email. I can provide an email, but then it asks for a phone number.

There are 100s of weather sources, some don't require a phone number, some do.

We already support 18 weather sources with NO mandatory phone number, including one source specific to China. Supporting all weather sources we currently support and that don't require a phone number is already very time-consuming, and no one is paying me to do that on my free time, so we have to make some choices because I'm not going to pay or give up my privacy for every weather company in the world. Additionally, users of Breezy Weather trust this app because I care about their privacy.

If privacy is not a criteria for you, you can use any other weather app, including the [QWeather app](https://www.qweather.com/en/app), or you can just fork this app (the [following tag](https://github.com/breezy-weather/breezy-weather/tree/archive/source-qweather) has an old version of Breezy Weather with support for QWeather, that you can just rebase on top of latest commit) ;)
